sectionate/gridutils.py

```diff
@@ -0,0 +1,95 @@
+    """
+    Find longitude and latitude coordinates from grid dataset, assuming the coordinate
+    names contain the sub-strings "lon" and "lat", respectively.
+
+    Parameters
+    ----------
+    grid: xgcm.Grid
+        Contains information about ocean model grid discretization, e.g. coordinates and metrics.
+        
+    Returns
+    -------
+    dict
+        Dictionary containing names of longitude and latitude coordinates.
+    """
+    dims = {}
+    for axis in ["X", "Y"]:
+        if "outer" in grid.axes[axis].coords:
+            dims[axis] = grid.axes[axis].coords["outer"]
+        elif "right" in grid.axes[axis].coords:
+            dims[axis] = grid.axes[axis].coords["right"]
+        else:
+            raise ValueError("Only 'symmetric' and 'non-symmetric' grids\
+            are currently supported. They require C-grid topology, i.e. with\
+            vorticity coordinates at 'outer' and 'right' positions, respectively.")
+
+    coords = grid._ds.coords
+    return {
+        axis: [
+            coords[c] for c in coords
+            if (
+                (geoc in c) and
+                (dims["X"] in coords[c].dims) and
+                (dims["Y"] in coords[c].dims)
+            )
+        ][0]
+        for axis, geoc in zip(["X", "Y"], ["lon", "lat"])
+    }
+    
+def coord_dict(grid):
+    """
+    Find names of "X" and "Y" dimension variables from grid dataset.
+
+    Parameters
+    ----------
+    grid: xgcm.Grid
+        Contains information about ocean model grid discretization, e.g. coordinates and metrics.
+        
+    Returns
+    -------
+    dict
+        Dictionary containing names of "X" and "Y" dimension variables, at both cell 'center' position
+        and either 'outer' or 'right' position. ('left' position not yet supported.)
+    """
+    if check_symmetric(grid):
+        q_pos = "outer"
+    else:
+        q_pos = "right"
+        
+    return {
+        "X": {
+            "h": grid.axes['X'].coords["center"],
+            "q": grid.axes["X"].coords[q_pos]},
+        "Y": {
+            "h": grid.axes['Y'].coords["center"],
+            "q": grid.axes["Y"].coords[q_pos]},
+    }
+    
+def check_symmetric(grid):
+    """
+    Check whether the horizontal ocean model grid is symmetric or not, according to MOM6 conventions.
+    Symmetric C-grids have tracers on (M,N) 'center' positions and vorticity on (M+1, N+1) 'outer' positions.
+    Non-symmetric C-grids instead have vorticity on (M,N) 'right' positions.
+
+    Parameters
+    ----------
+    grid: xgcm.Grid
+        Contains information about ocean model grid discretization, e.g. coordinates and metrics.
+        
+    Returns
+    -------
+    symmetric : bool
+        True if symmetric; False if non-symmetric.
+        
+    """
+    pos_dict = {
+        p : ((p in grid.axes['X'].coords) and (p in grid.axes['Y'].coords))
+        for p in ["outer", "right"]
+    }
+    if pos_dict["outer"]:
+        return True
+    elif pos_dict["right"]:
```

Comment:
That's correct. It does not yet support grids with "left" vorticity conventions, which would include ROMS. It should be straightforward to add another conditional for the "left" convention. I had not yet implemented it because I didn't have test data for it, but I just realized I can remove the "right" boundary from the symmetric MOM6 grid to generate a ROMS-like grid.

That said, the more robust approach moving forward would probably be to have a single algorithm for symmetric grids and instead just convert non-symmetric grids into symmetric ones by adding the padding implied by the BCs (periodic or no-normal-flow).

Reply:
If it's okay with you, I'd like to leave this for a future release since it's not yet tested and would require some substantial new code.